js/coinsbit.js

```diff
@@ -0,0 +1,799 @@
+        if (startTime === undefined) {
+            startTime = endTime - maxRequestedMilliSeconds;
+        } else {
+            endTime = this.sum (startTime, maxRequestedMilliSeconds);
+        }
+        startTime = startTime / 1000;
+        endTime = endTime / 1000;
+        request['start'] = parseInt (startTime);
+        request['end'] = parseInt (endTime);
+        response = await this.publicGetKline (this.extend (request, params));
+        // {
+        //     success: true,
+        //     message: "",
+        //     result: {
+        //         market: "BTC_USDT",
+        //         start: 1640000000,
+        //         end: 1640001234,
+        //         interval: 60,
+        //         kline: [
+        //             {
+        //                 time: 1640000000,
+        //                 open: "46979.18",
+        //                 close: "46893.88",
+        //                 highest: "46979.18",
+        //                 lowest: "46893.88",
+        //                 volume: "2.35947525",
+        //                 amount: "110745.7371631963",
+        //                 market: "BTC_USDT"
+        //             },
+        //             {
+        //                 time: 1640000060,
+        //                 ....
+        const result = this.safeValue (response, 'result', {});
+        const data = this.safeValue (result, 'kline', []);
+        return this.parseOHLCVs (data, market, timeframe, since, limit);
+    }
+
+    parseOHLCV (ohlcv, market = undefined) {
+        //
+        //
+        //     {
+        //         time: '1640051700',
+        //         open: '46954.3',
+        //         close: '46932.07',
+        //         highest: '46960.75',
+        //         lowest: '46932.07',
+        //         volume: '1.6985718',
+        //         amount: '79753.7310157163',
+        //         market: 'BTC_USDT'
+        //     }
+        //
+        // the ordering in swap / contract candles is OHLCV
+        //
+        return [
+            this.safeTimestamp (ohlcv, 'time'),
+            this.safeNumber (ohlcv, 'open'),
+            this.safeNumber (ohlcv, 'highest'),
+            this.safeNumber (ohlcv, 'lowest'),
+            this.safeNumber (ohlcv, 'close'),
+            this.safeNumber (ohlcv, 'volume'), // At this moment, we do not make any use of 'amount' property
+        ];
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'market': market['id'],
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit; // default = 50, minimum = 50, maximum = 1000
+        }
+        const defaultMethod = this.safeString (this.options, 'fetchTradesMethod');
+        const method = this.safeString (params, 'method', defaultMethod);
+        let data = {};
```

Comment:
Let's not initialize the variables to empty strings or empty objects:

```JavaScript
let data = undefined; // this is the way to go
```

It would be great if you could clean up all other occurrences of this type across the entire file.